src/shapes/Object/types/ObjectProps.ts

```diff
@@ -8,6 +10,7 @@ export interface ObjectProps extends SerializedObjectProps, ClipPathProps {
   fill: TFiller | string | null;
   stroke: TFiller | string | null;
   shadow: Shadow | null;
+  canvas?: StaticCanvas | Canvas;
```

Comment:
```suggestion
  canvas?: StaticCanvas;
```

And in `src/shapes/Object/types/FabricObjectProps.ts`
```ts
  canvas?: Canvas;
```